src/components/Carousel/Carousel.js

```diff
@@ -47,61 +44,85 @@ class Carousel extends Component {
   onResize() {
     this.resizeTheCarousel();
     this.checkNumOfSlidesToScroll();
+    this.getSlidesCount();
   }
 
-  resizeTheCarousel() {
-    const { widthOfSlide } = this.state;
-    let carouselWidth = '0';
-
-    if (window.innerWidth > 1280) {
-      carouselWidth = `${widthOfSlide * 5}px`;
-    } else if (window.innerWidth > 1024) {
-      carouselWidth = `${widthOfSlide * 4}px`;
-    } else if (window.innerWidth > 768) {
-      carouselWidth = `${widthOfSlide * 3}px`;
-    } else {
-      carouselWidth = `${widthOfSlide * 2}px`;
+  getSlidesCount = () => {
+    const sizeValues = {
+      desktop: 3,
+      tablet: 2,
+      mobile: 1
+    };
+
+    let count;
+
+    if (window.innerWidth >= 1024) {
+      count = sizeValues.desktop;
+    } else if (window.innerWidth >= 768) {
+      count = sizeValues.tablet;
+    } else if (window.innerWidth >= 320) {
+      count = sizeValues.mobile;
     }
 
-    this.carouselViewport.current.parentElement.style.width = carouselWidth;
+    return count;
+  };
+
+  resizeTheCarousel() {
+    this.setState({ slidesCount: this.getSlidesCount() });
+    this.countTheSlideWidth();
+  }
+
+  countTheSlideWidth() {
+    const { scrollWidth } = this.carouselViewport.current.parentElement;
+    const { slidesCount } = this.state;
+
+    this.setState({ widthOfSlide: scrollWidth / slidesCount });
   }
 
   checkIfSlidesAllTheWayOver() {
-    const { scrollLeft, clientWidth, scrollWidth } = this.carouselViewport.current;
+    const {
+      scrollLeft,
+      clientWidth,
+      scrollWidth
+    } = this.carouselViewport.current;
     // if scrollLeft == 0
     // hide left button
-    const allTHeWayLeftValue = (scrollLeft === 0);
+    const allTHeWayLeftValue = scrollLeft === 0;
 
     // if scrollLeft + viewPortOffset.length === whole viewPort length
     // 9 cards - each 120px: 9 * 210all === whole viewPort length
     // hide the rightScrollButton
-    const allTheWayRightValue = ((scrollLeft + clientWidth) === scrollWidth);
+    const allTheWayRightValue = scrollLeft + clientWidth === scrollWidth;
     const { allTheWayLeft, allTheWayRight } = this.state;
 
-    (allTheWayLeft !== allTHeWayLeftValue || allTheWayRight !== allTheWayRightValue)
+    (allTheWayLeft !== allTHeWayLeftValue
+      || allTheWayRight !== allTheWayRightValue)
       && this.setState({
         allTheWayLeft: allTHeWayLeftValue,
         allTheWayRight: allTheWayRightValue
       });
   }
 
   checkNumOfSlidesToScroll() {
-    const numOfSlidesToScroll = window.innerWidth <= 900 ? 2 : 4;
+    const numOfSlidesToScroll = window.innerWidth <= 900 ? 1 : 2;
```

Comment:
maybe this values should be equal to sizeValues count? As i understand here you're defining steps for prev and next buttons. If not- it's ok.

Reply:
Yes, I define here how many cards should be passed when button is clicked.

sizeValues count is made to show exact amount of cards I want to see according to screen width. 

Those are different things and have different values :)